cartographer_grpc/framework/rpc.cc

```diff
@@ -305,16 +326,23 @@ ActiveRpcs::~ActiveRpcs() {
 
 Rpc* ActiveRpcs::Add(std::unique_ptr<Rpc> rpc) {
   cartographer::common::MutexLocker locker(&lock_);
-  const auto result = rpcs_.emplace(rpc.release());
+  std::shared_ptr<Rpc> shared_rpc = std::move(rpc);
```

Comment:
I think the shared_ptr should be made by Rpc itself, see other comment.

Reply:
See my comment below.